# Construction of pYPK0_PGI1_ScXKS1_FBA1 (pMEC1129)

[pYPKa_Z_PGI1](pYPKa_Z_PGI1.ipynb)

[pYPKa_E_FBA1](pYPKa_E_FBA1.ipynb)

In [1]:
import pydna
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 = pydna.read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_PGI1 = pydna.read("pYPKa_Z_PGI1.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001181323.3) record

In [6]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169869 width=100% height=250></iframe>')

In [7]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001181323.3")

In [8]:
gene_template.list_features()

+----------+-----------+-------+-------+--------+--------------+--------+------+
| Feature# | Direction | Start |  End  | Length | id           | type   | orf? |
+----------+-----------+-------+-------+--------+--------------+--------+------+
| 0        |    -->    |   0   |  1803 |   1803 | <unknown id> | source | yes  |
| 1        |    -->    |   <0  | >1803 |   1803 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   |  1803 |   1803 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+-------+--------+--------------+--------+------+

In [9]:
orf = gene_template.extract_feature(2)

In [10]:
orf.isorf()

True

In [11]:
print(len(orf))
orf.seguid()

1803


27N2cQNKNSUXfhxo37o-d6AshX0

In [12]:
print(str(orf.seq.translate()))

MLCSVIQRQTREVSNTMSLDSYYLGFDLSTQQLKCLAINQDLKIVHSETVEFEKDLPHYHTKKGVYIHGDTIECPVAMWLEALDLVLSKYREAKFPLNKVMAVSGSCQQHGSVYWSSQAESLLEQLNKKPEKDLLHYVSSVAFARQTAPNWQDHSTAKQCQEFEECIGGPEKMAQLTGSRAHFRFTGPQILKIAQLEPEAYEKTKTISLVSNFLTSILVGHLVELEEADACGMNLYDIRERKFSDELLHLIDSSSKDKTIRQKLMRAPMKNLIAGTICKYFIEKYGFNTNCKVSPMTGDNLATICSLPLRKNDVLVSLGTSTTVLLVTDKYHPSPNYHLFIHPTLPNHYMGMICYCNGSLARERIRDELNKERENNYEKTNDWTLFNQAVLDDSESSENELGVYFPLGEIVPSVKAINKRVIFNPKTGMIEREVAKFKDKRHDAKNIVESQALSCRVRISPLLSDSNASSQQRLNEDTIVKFDYDESPLRDYLNKRPERTFFVGGASKNDAIVKKFAQVIGATKGNFRLETPNSCALGGCYKAMWSLLYDSNKIAVPFDKFLNDNFPWHVMESISDVDNENWDRYNSKIVPLSELEKTLI*


In [13]:
pf, pr = pydna.parse('''>XX_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTTGTGTTCAGTAATTCAGAG
                        >XK_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAGATGAGAGTCTTTTCCAGTTC''')

In [14]:
g = pydna.pcr( pf, pr, orf)

In [15]:
g.figure()

                                        5ATGTTGTGTTCAGTAATTCAGAG...GAACTGGAAAAGACTCTCATCTAA3
                                                                   |||||||||||||||||||||||| tm 55.3 (dbd) 60.9
                                                                  3CTTGACCTTTTCTGAGAGTAGATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTTGTGTTCAGTAATTCAGAG3
                                         ||||||||||||||||||||||| tm 54.4 (dbd) 59.0
                                        3TACAACACAAGTCATTAAGTCTC...CTTGACCTTTTCTGAGAGTAGATT5

In [16]:
terminator_clone = pYPKa_E_FBA1 = pydna.read("pYPKa_E_FBA1.gb")

In [17]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [18]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [19]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(1161), Amplicon(1881), Amplicon(736))

In [20]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [21]:
asm

Assembly:
Sequences........................: [5681] [1161] [1881] [736]
Sequences with shared homologies.: [5681] [1161] [736] [1881]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [9213]
Linear products..................: [9311] [9274] [9263] [9250] [8575] [8200] [7419] [6744] [6356] [3691] [2992] [2580] [98] [61] [50] [37]

In [22]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|1161bp_PCR_prod|38
|                           \/
|                           /\
|                           38|1881bp_PCR_prod|35
|                                              \/
|                                              /\
|                                              35|736bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [23]:
pYPK0_PGI1_ScXKS1_FBA1 = candidate.synced(pYPK0)

The pYPK0_PGI1_ScXKS1_FBA1 has cseguid ```tDAwCS0RGOllmBII5oQTORuFXpw``` and 9213 bp.

In [24]:
print(len(pYPK0_PGI1_ScXKS1_FBA1))
pYPK0_PGI1_ScXKS1_FBA1.cseguid()

9213


tDAwCS0RGOllmBII5oQTORuFXpw

In [25]:
pYPK0_PGI1_ScXKS1_FBA1.name="pMEC1129"
#pYPK0_PGI1_ScXKS1_FBA1.description = "pYPK0_PGI1_ScXKS1_FBA1tp"
pYPK0_PGI1_ScXKS1_FBA1.stamp()

cSEGUID_tDAwCS0RGOllmBII5oQTORuFXpw

In [26]:
pYPK0_PGI1_ScXKS1_FBA1.write("pYPK0_PGI1_ScXKS1_FBA1.gb")

pYPK0_PGI1_ScXKS1_FBA1.gb

##Download sequence

[pYPK0_PGI1_ScXKS1_FBA1](pYPK0_PGI1_ScXKS1_FBA1.gb)

In [27]:
reloaded = pydna.read("pYPK0_PGI1_ScXKS1_FBA1.gb")

In [28]:
reloaded.cseguid()

tDAwCS0RGOllmBII5oQTORuFXpw

In [29]:
reloaded.description

'cSEGUID_tDAwCS0RGOllmBII5oQTORuFXpw_2015-07-15T12:34:16.545087'